### PR TITLE
Eve commit

### PR DESCRIPTION
**Pull request by RudolphRiedel**
For the most part this includes a reworked init().
I started adding more settings and changing the names to reflect modules instead just resolutions.
FTDI_FT800CB and FTDI_4DLCD_FT843 are almost the same.
But FTDI_4DLCD_FT843 does use GPIO_1 for amplifier control while FT800CB does not.
And the resistive touch from HAOYU displays is rather cheap so it needs a higher sensitivity than quality modules like the ones from FTDI or 4D Systems.

The distinction between FT800 and FT810 also is misleading.
It is FT80x and FT81x really.
But I did not touch that.

I have my spare CR-10 controller board with the FT800CB up and running right now on my desk.
Well, technically it is a FT810CB board with the panel from a FT800CB.

I have not checked into higher level stuff but as it still flickers from time to time and still gets stuck when playing with the display and still sends data to the display without a noticeable pause, the interface layer holds some bugs to be fixed.

### CODE REVIEW

File: Marlin/src/extensible-ui/lib/ftdi_eve_spi.cpp [hunk 1]
```diff
@@ -153,9 +152,29 @@ void CLCD::test_pulse(void)
   }
 #endif
 
+#if defined(CLCD_USE_SOFT_SPI)
+  void CLCD::_soft_spi_send (uint8_t spiOutByte) {
```
Comment (marcio-ao) [inline Marlin/src/extensible-ui/lib/ftdi_eve_spi.cpp:156]: Why did you add this? _soft_spi_transfer(spiOutByte) does the same thing if you ignore the return value.

**Review by marcio-ao — APPROVED** [review]

File: Marlin/src/extensible-ui/lib/ftdi_eve_spi.cpp (hunk 1 above)
Comment (RudolphRiedel) [reply Marlin/src/extensible-ui/lib/ftdi_eve_spi.cpp:156]: While the result is the same it is several cycles faster when it does not need to read the pin in the first place and therefore the soft-spi as a whole is faster.
I clocked it before at 1MHz and with the modification at 1,3MHz.
Not much overall but 30% faster.
And >95% of all spi transactions are write-only.

Comment (marcio-ao) [reply Marlin/src/extensible-ui/lib/ftdi_eve_spi.cpp:156]: That makes sense. By the way, I am working on a new commit that cleans up some confusion in the code. I realized that there was a need to distinguish between which panel was being used, as well as which pins were being used to interface the panel with the controller board.

The previous code didn't make this distinction clear since all our printers are using the same board.

When you added a config for the CR-10, the need for a separation became clear. I am working on a commit that addresses that.
